Sources/Signal.swift

```diff
@@ -1484,6 +1484,32 @@ extension Signal {
 	public func debounce(_ interval: TimeInterval, on scheduler: DateScheduler) -> Signal<Value, Error> {
 		return flatMapEvent(Signal.Event.debounce(interval, on: scheduler))
 	}
+
+	/// Starts delaying values on `scheduler` if not enough `interval`
+	/// seconds have passed since `self` last sent a value.
+	///
+	/// If at least `interval` seconds have passed since `self` last sent a
+	/// value, then the value is going to be forwarded on the given scheduler
+	/// without any delay. You can look at it as a minimum outgoing rate.
+	///
+	/// - seealso: `throttle`, `delay`
+	///
+	/// - note: If multiple values are received before the `interval` has
+	///         elapsed, these values are going to be delayed one by one until
+	///         the `interval` value elapses, until there's no more queued
+	///         values to be sent.
+	///
+	/// - precondition: `interval` must be non-negative number.
+	///
+	/// - parameters:
+	///   - interval: Number of seconds to enqueue sent values.
+	///   - scheduler: A scheduler to send values on.
+	///
+	/// - returns: A signal that sends values with a minimun of `interval`
+	///            seconds appart from each other on a given scheduler.
+	public func bottlenecked(_ interval: TimeInterval, on scheduler: DateScheduler) -> Signal<Value, Error> {
```

Comment:
It seems that it is similar to `throttle`, just that it queues all instead of just the latest occurrence. I wonder if we should extend `throttle` to have a `ThrottleStrategy` instead.

Reply:
I don't think it fits into it. Even though it has a similar implementation, it has a different behaviour which makes it a different concept from what we're expecting from a throttle. If the implementation was completely different (actually lets forget it), would the concept still fit into throttle? 
Thoughts?

Reply:
That could make sense. We could include `debounce` at the same time. All 3 are conceptually similar, but have a different strategy. I think our grouping of operators has been very successful and the existing `debounce`/`throttle` divide is very confusing.

The tricky bit is coming up with good names for the strategies. `queue` or `buffer` could work for the `bottlenecked` case. But how do you succinctly describe the existing `throttle` and `dequeue`? I like describing them as "time since last value was received by `self`" and "time since last value was sent by `self`", but those aren't succinct enough.

**Regardless**, I think the `bottlenecked` name isn't good enough. It doesn't really capture the behavior of the operator (but it makes sense if you already understand what the operator does). I think we have the strongest argument for inclusion/acceptance when it fits into an existing name like `throttle`, but we have to find something that works.

Reply:
Regarding the bottlenecked method name, if we play around with the name of the parameter we can use other alternatives:

>bottleneck(_ throughput: TimeInterval, on scheduler: DateScheduler)
>buffer(_ deflateOneAfter: TimeInterval, on scheduler: DateScheduler)
>constraint(_ minimumSpread: TimeInterval, on scheduler: DateScheduler)

Now finding a good names for throttle strategies that shortly describe the behaviour is more complicated. Here's a suggestion that makes sense together:
>Throttle: ThrottleStrategy.beat since the speed at which the constraint (time interval) runs sets the "beat"
>Debounce: ThrottleStrategy.fluctuate since its last value received that starts the constraint
>Bottlenecked: ThrottleStrategy.buffer or ThrottleStrategy.queue or ThrottleStrategy.enqueue or ThrottleStrategy.spread

Reply:
The best I can think of would be:

* `.queued`
* `.fromEarliest`
* `.fromLatest`

Reply:
What if the master operator was `regulate(over interval: TimeInterval, using strategy: RegulationStrategy, on scheduler: DateScheduler)`?

Where the options for `RegulationStrategy` were:
 - `.throttle`
 - `.debounce`
 - `.bottleneck`
 - `.suppress` 

(`.suppress` is a new strategy I'm proposing in an upcoming PR that will forward the latest value on a scheduler immediately then wait until an interval has passed before sending more values; during the interval, values are suppressed and will not be forwarded after the interval expires. Each new value resets the duration.)

Reply:
The base name makes sense, since it is more than just time shifting. `delay` could fall under this too.

Reply:
I would prefer to have the strategy first though, akin to `flatMap`.

Reply:
I think `throttle` makes the most sense as an operator name. And I dislike `throttle` and `debounce` as strategy names because I can never remember which is which.

Reply:
Hmm, what about `throttle` as the base name, and the `elapse`, `defer` and `buffer`?

Reply:
We're getting somewhere :)
To give some space for the to be implemented `suppress` strategy (and others), should we rename `all` to `queued` instead?

Reply:
I think `queue` or `queued` is a good name.

`skipPrevious` could apply to either `throttle` or `debounce`, so I'm not wild about it.

Reply:
It is quite hard to be concise without bringing in new verbs. Like grace, silence, coalescing+distancing, etc.

It does make me wonder if we express them in combinations though. Say `.distance(.all)` being the bottleneck strategy proposed here, while `.distance(.latest)` being the current `throttle`. `debounce` can be `.silence(.latest)`, and `.silence(.all)` would be effectively `defer`.

Reply:
I think `throttle` would make sense as the base name. 

> Throttle: a device controlling the flow of fuel or power to an engine.

Reply:
I was talking about `distance` and `silence` as strategy names with `throttle` as the base name.

Reply:
@andersio ah, I see. I didn't realize `.all`, `.latest` were parameters to the strategy.

Reply:
I like that approach of just using throttle as the base operator with the combinations for the different strategies. 
I'll try to update this soon, although under a bit of a squeeze for time. Since there's a throttle(while:) option and only one implementation, Im going to need to add support for this option to the missing methods.